app/controllers/provider/admin/cms/templates_controller.rb

```diff
@@ -31,7 +31,18 @@ def update
     @page.build_version if params[:version]
 
     if @page.save
-      success_update
+      msg = publish_or_hide_page_with_message
+      respond_to do |format|
+        format.html do
+          flash[:notice] = msg
+          redirect_to(action: :edit, id: @page.id)
+        end
+
+        format.js do
+          flash.now[:notice] = msg
+          render template: '/provider/admin/cms/templates/update'
+        end
+      end
     else
       respond_to do |format|
         format.html do
```

Comment:
For consistency, this could be a one liner
```ruby
format.html { render :edit }
```

Reply:
Done